src/backend/executor/process.ts

```diff
@@ -0,0 +1,147 @@
+import * as child_process from 'child_process';
+import * as path from 'path';
+import { Event, EventEmitter, ExtensionContext, Uri, workspace } from 'vscode';
+
+export enum ProcessStatus {
+    notRunning,
+    running,
+    killed,
+    finished,
+    errored,
+}
+
+export class ExecutorProcess {
+    activeProcess?: child_process.ChildProcess;
+
+    /** Every line should have a newline at the end */
+    private _processStdout: EventEmitter<string> = new EventEmitter();
+    /**
+     * Standard output of the process.
+     *
+     * Also contains other metadata:
+     * * ``> command`` for each executed command,
+     * * ``>>> metadata`` for other information, eg. when a process is finished.
+     */
+    public get processStdout(): Event<string> {
+        return this._processStdout.event;
+    }
+
+    /** Every line should have a newline at the end */
+    private _processStderr: EventEmitter<string> = new EventEmitter();
+    /** Standard error of the process. */
+    public get processStderr(): Event<string> {
+        return this._processStderr.event;
+    }
+
+    private _processStatus: ProcessStatus = ProcessStatus.notRunning;
+    public get processStatus(): ProcessStatus {
+        return this._processStatus;
+    }
+
+    private _processStatusChange: EventEmitter<ProcessStatus> = new EventEmitter();
+    public get processStatusChange(): Event<ProcessStatus> {
+        return this._processStatusChange.event;
+    }
+
+    constructor(_ctx: ExtensionContext) {
+    }
+
+    public getProcessCmdLine(...files: Uri[]): string {
+        if (this.activeProcess !== undefined) {
+            // TODO: Better error handling
+            return '';
+        }
+
+        if (!workspace.workspaceFolders?.length) {
+            return '';
+        }
+
+        const workspaceFolder = workspace.workspaceFolders[0].uri.fsPath;
+
+        const getConfigAndReplaceVariables = (category: string, name: string): string | undefined => {
+            const configValue = workspace.getConfiguration(category).get<string>(name);
+            return configValue
+                ?.replace(/\${workspaceRoot}/g, workspaceFolder)
+                .replace(/\${workspaceFolder}/g, workspaceFolder)
+                .replace(/\${cwd}/g, process.cwd())
+                .replace(/\${env\.([^}]+)}/g, (sub: string, envName: string) => process.env[envName] ?? '');
+        };
+
+        const ccPath = getConfigAndReplaceVariables('codechecker.executor', 'executablePath')
+            ?? 'codeChecker';
+        const ccFolder = getConfigAndReplaceVariables('codechecker.backend', 'outputFolder')
+            ?? path.join(workspaceFolder, '.codechecker');
+        const ccCompileCmd = path.join(ccFolder, 'compile_cmd.json');
```

Comment:
From what I've seen, `CodeChecker check`'s automatically created copy of the database is called `compile_cmd.json`.  
I'd like to rely on that as the default, but I can add more fallback options to locate the database inside the CodeChecker metadata directory.

Selecting an arbitrary compilation database will be added in a separate PR, when one is selected, that will be used instead of this fallback path.